### DISCUSSION TITLE
hot reloading with tower-livereload

### DISCUSSION THREAD

**Discussion by lmezg** [salvo-rs/salvo#457] [Q&A] [answered]
Hi
I noticed tower layer is supported by the new release! Thanks for the hard works! I'm having some issues using tower-livereload:
````rust
use salvo::prelude::*;
use tokio::time::Duration;
use tower_livereload::LiveReloadLayer;

#[handler]
async fn hello() -> &'static str {
    "Hello World"
}

#[tokio::main]
async fn main() {
    tracing_subscriber::fmt().init();

    let hotreload = LiveReloadLayer::new().compat();
    let acceptor = TcpListener::new("0.0.0.0:5800").bind().await;
    let router = Router::new().hoop(hotreload).get(hello);
    Server::new(acceptor).serve(router).await;
}
````
Compiler complains 
````
 the trait `http_body::Body` is not implemented for `ResBody`
````
is there a trait i need to included ? I assume LiveReloadLayer is just a normal tower layer and it should work with salve. Thanks

**Comment by chrislearn** [comment, marked as answer]
Salvo is base on `hyper` 1.0 rc4, it use `http-body` is version 1.0.0-rc2, but `tower-livereload` use `http-body` version is 0.4.5.

**Reply by lmezg** [reply]
Thanks for checking!
